lms/views/dashboard/api/user.py

```diff
@@ -131,30 +132,33 @@ def students_metrics(self) -> APIStudents:
         for user in self.request.db.scalars(users_query).all():
             if s := stats_by_user.get(user.h_userid):
                 # We seen this student in H, get all the data from there
-                students.append(
-                    APIStudent(
-                        h_userid=user.h_userid,
-                        lms_id=user.user_id,
-                        display_name=s["display_name"],
-                        annotation_metrics=AnnotationMetrics(
-                            annotations=s["annotations"] + s["page_notes"],
-                            replies=s["replies"],
-                            last_activity=s["last_activity"],
-                        ),
-                    )
+                api_student = APIStudent(
+                    h_userid=user.h_userid,
+                    lms_id=user.user_id,
+                    display_name=s["display_name"],
+                    annotation_metrics=AnnotationMetrics(
+                        annotations=s["annotations"] + s["page_notes"],
+                        replies=s["replies"],
+                        last_activity=s["last_activity"],
+                    ),
                 )
             else:
                 # We haven't seen this user H,
                 # use LMS DB's data and set 0s for all annotation related fields.
-                students.append(
-                    APIStudent(
-                        h_userid=user.h_userid,
-                        lms_id=user.user_id,
-                        display_name=user.display_name,
-                        annotation_metrics=AnnotationMetrics(
-                            annotations=0, replies=0, last_activity=None
-                        ),
-                    )
+                api_student = APIStudent(
+                    h_userid=user.h_userid,
+                    lms_id=user.user_id,
+                    display_name=user.display_name,
+                    annotation_metrics=AnnotationMetrics(
+                        annotations=0, replies=0, last_activity=None
+                    ),
                 )
 
+            if assignment.auto_grading_config:
+                api_student["auto_grading_grade"] = calculate_grade(
```

Comment:
Calculating the value on the fly should be fine for now.

If at some point we want to track which grades have already been synced and which aren't, we'll probably have to persist pre-calculated grades somehow.

Reply:
Yes, we'd need to store them to support the "NEW" badge.